### ISSUE TITLE
module 'oemof' has no attribute '__version__' after installing oemof.db

### ISSUE THREAD

**Issue by livhendx** [oemof/oemof#269]
I can't run the examples anymore, because of the following error:

module `oemof` has no attribute `__version__`

I was able to run them a week ago. I did update spyder. I reinstalled oemof, but without any effect...

If I comment logger.define_logging(), I get the following error:
PermissionError: [Errno 13] Permission denied: 'C:\Users\...\.oemof\lp_files\storage_invest.lp'

Do I get these errors because of spyder-update?
Or how can I work around these issues?

Thanks!

**Comment by uvchik** [comment]
It does not have to do anything with Spyder.

If you are not sure whether it is a Spyder problem you can just open a prompt and type:
```python
import oemof
oemof.__version__
```
Your first problem is caused by the installation of oemof.db but I don't know why. We will try to fix it.

You can comment the import of the logger and the call of the logger. To get some logging messages you can add a basic logger:
```python
# from oemof.tools import logger
[...]
# logger.define_logging()
logging.basicConfig(level='INFO')
```

```PermissionErrors``` normally belong to your operating system (Windows) but sometimes they are not real permission errors. Try to find the path on your system. Use the following lines if you do not know how.
```python
import os
print(os.path.expanduser('~'))
```

There should be a folder named `.oemof`. In this folder should be the folder `lp_files`. You should have writing access to both folders.

**Comment by uvchik** [comment]
@gnn : To you have an idea why `oemof.__version__` does not work after installing `oemof.db`?

**Comment by livhendx** [comment]
hi,
I could work around the logger.deinfe_logging() issue.
And I changed the rights of .oemof folder.

So... the examples are working again!

Thank you very much!!

**Comment by uvchik** [comment]
@livhendx : You are welcome. We are always interested in projects using oemof. Aims, problems, feedback. We would appreciate a short description to (https://oemof.wordpress.com/contact/). Thanks in advance.
@gnn : We still should solve this issue.

**Comment by gnn** [comment]
Yeah, you're right @uvchik, we should solve this. This one has been open for far too long. I'll gather the issues pertaining to this until Friday and consolidate them here. After this, I'll try to fix the issue starting on Tuesday the 21st. I should be able to either fix this till the 24th, or know more about why this happens.

**Comment by uvchik** [comment]
Okay, if we will not solve this until the end of February I will publish a workaround.

**Comment by gnn** [comment]
Ok. I seem to finally have a grip on this. I was able reproduce the error with this specific scenario:

  - install `oemof.db` first,
  - then install `oemof` as a an editable install using `pip install -e`.

After that, getting the version number no longer works. I added a test for this, and a `tox.ini` file so that we can use [tox](https://tox.readthedocs.io/en/latest/) to run our tests in various configurations on the [fixes/#269](/oemof/oemof/tree/fixes/%23269) branch. If you check out that branch, `pip install tox`, which you can do in a virtual environment, and then run `tox`, you should see that the `db-first-dev` environment fails due to being unable to retrieve `oemof.__version__`.
What happens here is the following:

  - Both, `oemof` and `oemof.db` contain an `__init__.py` file under the `oemof` directory.
  - Pyhon **only** loads the first of these files it encounters, when instructed to `import oemof`.
  - If `oemof` is (re)installed on top of an already existing `oemof.db` installation, the latter's `__init__.py` is the older one and thus probably takes precedence over the former one's when trying to `import oemof`.
  - Since this `__init__.py` does not export a `__version__` attribute, Python can't find one.

The immediate fix I tried, which worked and didn't break the few things I tried in `oemof.db`, was removing the `__init__.py` file from `oemof.db`. This was added by @uvchik though (see #78 and oemof/oemof.db@01008aed1d3a59d40e51d48efae1643392d6125d) so I don't know what can break if we just go ahead and remove that.
Long story short: I think this is an issue which has to be resolved in `oemof.db`, so it doesn't pertain to an `oemof` release. I'd like to solidify this hypothesis using further tests in a few hours, but they are not strictly necessary.

**Comment by gnn** [comment]
So I made a [branch](/oemof/oemof.db/tree/fixes/oemof-%23269) where there is no `oemof/db/__init__.py` file in `oemof.db` and there are minimal tests to check that `oemof.db` is still importable. I tested both, [fixes/#269](/oemof/oemof/tree/fixes/%23269) and the branch mentioned above against each other and things seem to work.
I'd like somebody else to have a look at this, but if everybody else concurs, I think we can open a PR in `oemof.db` and close this with a reference to the PR.

**Comment by uvchik** [comment]
@gnn Thank you for fixing this bug. This was really annoying.
